### PR TITLE
Add beta regional resources to composite

### PR DESCRIPTION
**Pull request by spencerhance**
This will be necessary to use L7-ILB Beta

### CODE REVIEW

**Comment by rramkumar1** [comment]
@spencerhance Is this ready for review? Not sure since its not explicitly assigned.

**Comment by spencerhance** [comment]
@rramkumar1 whoops my bad.  /assign @rramkumar1 😄

File: pkg/composite/gen/main.go [hunk 1]
```diff
@@ -507,8 +516,14 @@ func Update{{.Name}}(gceCloud *gce.Cloud, key *meta.Key, {{.VarName}} *{{.Name}}
...
 			return err
 		}
-		klog.V(3).Infof("Updating beta {{.Name}} %v", beta.Name)
-		return mc.Observe(gceCloud.Compute().Beta{{.GetCloudProviderName}}().Update(ctx, key, beta))
+		switch key.Type() {
+		case meta.Regional:
+		  klog.V(3).Infof("Updating beta region {{.Name}} %v", beta.Name)
+			return mc.Observe(gceCloud.Compute().BetaRegion{{.GetCloudProviderName}}().Update(ctx, key, beta))
+		default:
+      klog.V(3).Infof("Updating beta {{.Name}} %v", beta.Name)
```
Comment (bowei) [inline pkg/composite/gen/main.go]: fix tabbing

File: pkg/composite/gen/main.go [hunk 2]
```diff
@@ -536,8 +551,14 @@ func Delete{{.Name}}(gceCloud *gce.Cloud, key *meta.Key, version meta.Version) e
...
 		}
 	case meta.VersionBeta:
-		klog.V(3).Infof("Deleting beta {{.Name}} %v", key.Name)
-		return mc.Observe(gceCloud.Compute().Beta{{.GetCloudProviderName}}().Delete(ctx, key))
+		switch key.Type() {
+		case meta.Regional:
+		  klog.V(3).Infof("Deleting beta region {{.Name}} %v", key.Name)
+			return mc.Observe(gceCloud.Compute().BetaRegion{{.GetCloudProviderName}}().Delete(ctx, key))
+		default:
+		  klog.V(3).Infof("Deleting alpha {{.Name}} %v", key.Name)
```
Comment (bowei) [inline pkg/composite/gen/main.go]: alpha => beta

File: pkg/composite/gen/main.go [hunk 3]
```diff
@@ -562,8 +583,14 @@ func Get{{.Name}}(gceCloud *gce.Cloud, key *meta.Key, version meta.Version) (*{{
...
 		}
 	case meta.VersionBeta:
-    klog.V(3).Infof("Getting beta {{.Name}} %v", key.Name)
-		gceObj, err = gceCloud.Compute().Beta{{.GetCloudProviderName}}().Get(ctx, key)
+		switch key.Type() {
+			case meta.Regional:
+		  	klog.V(3).Infof("Getting beta region {{.Name}} %v", key.Name)
+				gceObj, err = gceCloud.Compute().BetaRegion{{.GetCloudProviderName}}().Get(ctx, key)
+			default:
+		  	klog.V(3).Infof("Getting alpha {{.Name}} %v", key.Name)
```
Comment (bowei) [inline pkg/composite/gen/main.go]: alpha => beta

File: pkg/composite/gen.go [hunk 4]
```diff
@@ -3072,8 +3116,14 @@ func DeleteHealthCheck(gceCloud *gce.Cloud, key *meta.Key, version meta.Version)
...
 		}
 	case meta.VersionBeta:
-		klog.V(3).Infof("Deleting beta HealthCheck %v", key.Name)
-		return mc.Observe(gceCloud.Compute().BetaHealthChecks().Delete(ctx, key))
+		switch key.Type() {
+		case meta.Regional:
+			klog.V(3).Infof("Deleting beta region HealthCheck %v", key.Name)
+			return mc.Observe(gceCloud.Compute().BetaRegionHealthChecks().Delete(ctx, key))
+		default:
+			klog.V(3).Infof("Deleting alpha HealthCheck %v", key.Name)
```
[the remark this answers is not in the document]
Comment (spencerhance) [reply pkg/composite/gen.go]: Done

File: pkg/composite/gen.go [hunk 5]
```diff
@@ -2601,8 +2620,14 @@ func GetBackendService(gceCloud *gce.Cloud, key *meta.Key, version meta.Version)
...
 		}
 	case meta.VersionBeta:
-		klog.V(3).Infof("Getting beta BackendService %v", key.Name)
-		gceObj, err = gceCloud.Compute().BetaBackendServices().Get(ctx, key)
+		switch key.Type() {
+		case meta.Regional:
+			klog.V(3).Infof("Getting beta region BackendService %v", key.Name)
+			gceObj, err = gceCloud.Compute().BetaRegionBackendServices().Get(ctx, key)
+		default:
+			klog.V(3).Infof("Getting alpha BackendService %v", key.Name)
```
[the remark this answers is not in the document]
Comment (spencerhance) [reply pkg/composite/gen.go]: Done

File: pkg/composite/gen.go [hunk 6]
```diff
@@ -2575,8 +2588,14 @@ func DeleteBackendService(gceCloud *gce.Cloud, key *meta.Key, version meta.Versi
...
 		}
 	case meta.VersionBeta:
-		klog.V(3).Infof("Deleting beta BackendService %v", key.Name)
-		return mc.Observe(gceCloud.Compute().BetaBackendServices().Delete(ctx, key))
+		switch key.Type() {
+		case meta.Regional:
+			klog.V(3).Infof("Deleting beta region BackendService %v", key.Name)
+			return mc.Observe(gceCloud.Compute().BetaRegionBackendServices().Delete(ctx, key))
+		default:
+			klog.V(3).Infof("Deleting alpha BackendService %v", key.Name)
```
[the remark this answers is not in the document]
Comment (spencerhance) [reply pkg/composite/gen.go]: Done

File: pkg/composite/gen/main.go (hunk 1 above)
Comment (spencerhance) [reply pkg/composite/gen/main.go]: Done

File: pkg/composite/gen/main.go (hunk 2 above)
Comment (spencerhance) [reply pkg/composite/gen/main.go]: Done

File: pkg/composite/gen/main.go (hunk 3 above)
Comment (spencerhance) [reply pkg/composite/gen/main.go]: Done

**Comment by spencerhance** [comment]
@bowei this should be good to go now that k8s-cloud-provider has been updated

**Comment by bowei** [comment]
some of the indentation of the code templates have strange indentation. not going to block on that though.

/lgtm
/approve

**Pull request merged**